internal/common/kafka/kafka.go

```diff
@@ -18,26 +18,74 @@ var defaultTopic = "__consumer_offsets"
-		"retry.backoff.ms":         config.GetInt("kafka.retry.backoff.ms"),
-	})
+
+	var producer *kafka.Producer
+	var err error
+	if config.Get("kafka.sasl.username") != nil {
+		producer, err = kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers":        config.GetString("kafka.bootstrap.servers"),
+			"request.required.acks":    config.GetInt("kafka.request.required.acks"),
+			"message.send.max.retries": config.GetInt("kafka.message.send.max.retries"),
+			"retry.backoff.ms":         config.GetInt("kafka.retry.backoff.ms"),
+			"sasl.username":            config.GetString("kafka.sasl.username"),
+			"sasl.password":            config.GetString("kafka.sasl.password"),
+			"sasl.mechanism":           config.GetString("kafka.sasl.mechanism"),
+			"security.protocol":        config.GetString("kafka.sasl.protocol"),
+			"ssl.ca.location":          config.GetString("kafka.capath"),
+		})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		producer, err = kafka.NewProducer(&kafka.ConfigMap{
+			"bootstrap.servers":        config.GetString("kafka.bootstrap.servers"),
+			"request.required.acks":    config.GetInt("kafka.request.required.acks"),
+			"message.send.max.retries": config.GetInt("kafka.message.send.max.retries"),
+			"retry.backoff.ms":         config.GetInt("kafka.retry.backoff.ms"),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return producer, nil
 }
 
 func NewConsumer(ctx context.Context, config *viper.Viper, topic string) (*kafka.Consumer, error) {
-	consumer, err := kafka.NewConsumer(&kafka.ConfigMap{
-		"bootstrap.servers":        config.GetString("kafka.bootstrap.servers"),
-		"group.id":                 config.GetString("kafka.group.id"),
-		"auto.offset.reset":        config.GetString("kafka.auto.offset.reset"),
-		"auto.commit.interval.ms":  config.GetInt("kafka.auto.commit.interval.ms"),
-		"go.logs.channel.enable":   true,
-		"allow.auto.create.topics": true,
-	})
 
-	if err != nil {
-		return nil, err
+	var consumer *kafka.Consumer
+	var err error
+	if config.Get("kafka.sasl.username") != nil {
+		consumer, err = kafka.NewConsumer(&kafka.ConfigMap{
+			"bootstrap.servers":        config.GetString("kafka.bootstrap.servers"),
+			"sasl.username":            config.GetString("kafka.sasl.username"),
+			"sasl.password":            config.GetString("kafka.sasl.password"),
+			"sasl.mechanism":           config.GetString("kafka.sasl.mechanism"),
+			"security.protocol":        config.GetString("kafka.sasl.protocol"),
+			"ssl.ca.location":          config.GetString("kafka.capath"),
+			"group.id":                 config.GetString("kafka.group.id"),
+			"auto.offset.reset":        config.GetString("kafka.auto.offset.reset"),
+			"auto.commit.interval.ms":  config.GetInt("kafka.auto.commit.interval.ms"),
+			"go.logs.channel.enable":   true,
+			"allow.auto.create.topics": true,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		consumer, err = kafka.NewConsumer(&kafka.ConfigMap{
```

Comment:
same as above

Reply:
Thanks! Yeah, i was going back and forth over this one. I'll do it the other way.